### PR TITLE
Support individual shard repositioning

### PR DESCRIPTION
**Pull request by kriskowal**
- [x] Add iHops to EgressNodes computations
- [x] Thread iHops configuration through remote config
- [ ] Test

### CODE REVIEW

**Comment by kriskowal** [comment]
r @jcorbin @rf

File: clients/index.js [hunk 1]
```diff
@@ -671,6 +671,11 @@ ApplicationClients.prototype.updateKValues = function updateKValues(hasChanged,
         }
     }
 
+    if (forceUpdate || hasChanged['iHops.services']) {
```
Comment (jcorbin) [inline clients/index.js]: sUch cApitalization iS oNly aSking fOr pRoblems

File: egress-nodes.js [hunk 2]
```diff
@@ -40,6 +40,7 @@ function EgressNodes(options) {
     self.defaultKValue = options.defaultKValue;
 
     self.kValueForServiceName = {};
+    self.iHopsForServiceName = {}; // iHopsForServiceName[serviceName][i] -> i'
```
Comment (jcorbin) [inline egress-nodes.js]: `i'` for "eye prime" is a bit opaque as far as comments go... but sure

File: egress-nodes.js [hunk 3]
```diff
@@ -88,9 +97,13 @@ EgressNodes.prototype.exitsFor = function exitsFor(serviceName) {
     );
 
     var k = self.kValueFor(serviceName);
+    var iHops = self.iHopsFor(serviceName);
     // Object<hostPort: String, Array<lookupKey: String>>
     var exitNodes = Object.create(null);
     for (var i = 0; i < k; i++) {
+        if (iHops[i]) {
```
Comment (jcorbin) [inline egress-nodes.js]: ` !== undefined`, or do you mean to disallow mapping to `0`?

File: egress-nodes.js [hunk 4]
```diff
@@ -117,8 +130,12 @@ EgressNodes.prototype.isExitFor = function isExitFor(serviceName) {
     );
 
     var k = self.kValueFor(serviceName);
+    var iHops = self.iHopsFor(serviceName);
     var me = self.ringpop.whoami();
     for (var i = 0; i < k; i++) {
+        if (iHops[i]) {
```
Comment (jcorbin) [inline egress-nodes.js]: same wrt `0`

File: egress-nodes.js (hunk 3 above)
Comment (rf) [inline egress-nodes.js]: we can't map to 0 (or anything < k) but it's not purposeful

**Comment by rf** [comment]
lgtm

**Comment by jcorbin** [comment]
I'm 🆗 with this as long as @kriskowal  agrees wrt zaro.

**Comment by kriskowal** [comment]
I do agree wrt zaro hops. moar hops oar bettar. you’d never hop to a value less than k in practice, since it would just double up on a host that's already affine. However, there’s no harm in erring on the side of caution, so I’ve made the change.

**Comment by jcorbin** [comment]
Your battling for zaro hath inspired an LG™

**Pull request merged**